modules/jenkins-plugin/src/main/java/com/smartbear/ready/jenkins/RemoteTestStarter.java

```diff
@@ -0,0 +1,539 @@
+    private final String userName;
+    private final String password;
+    private final String repositoryName;
+    private final String environment;
+    private final String hostAndPort;
+    private final String tags;
+    private final String projectFilePassword;
+    private final String projectProperties;
+    private final String extraFiles;
+    private final String reportFile;
+    private File workspace;
+
+    // Fields in config.jelly must match the parameter names in the "DataBoundConstructor"
+    @DataBoundConstructor
+    public RemoteTestStarter(String pathToProjectFile,
+                             String testType,
+                             String serverUrl,
+                             String userName,
+                             String password,
+                             String testCaseName,
+                             String testSuiteName,
+                             String repositoryName, String environment,
+                             String hostAndPort, String tags,
+                             String projectFilePassword,
+                             String projectProperties, String extraFiles, String reportFile) {
+        this.pathToProjectFile = pathToProjectFile;
+        this.testType = testTypeForString(testType);
+        this.serverUrl = serverUrl;
+        this.userName = userName;
+        this.password = password;
+        this.testCaseName = testCaseName;
+        this.testSuiteName = testSuiteName;
+        this.repositoryName = repositoryName;
+        this.environment = environment;
+        this.hostAndPort = hostAndPort;
+        this.tags = tags;
+        this.projectFilePassword = projectFilePassword;
+        this.projectProperties = projectProperties;
+        this.extraFiles = extraFiles;
+        this.reportFile = reportFile;
+    }
+
+    private TestType testTypeForString(String typeString) {
+        return Arrays.stream(TestType.values()).filter(type -> type.name().equals(typeString))
+                .findFirst()
+                .orElse(TestType.LOCAL);
+    }
+
+    @Override
+    public boolean perform(AbstractBuild build, Launcher launcher, BuildListener listener) throws AbortException {
+        try {
+            workspace = new File(build.getWorkspace().toURI());
+        } catch (Exception e) {
+            throw new AbortException("Could not read path to workspace");
+        }
+        try {
+            List<TestServerExecution> execution = testType == TestType.REPOSITORY ? runRepositoryTests() :
+                    runTestFiles(build);
+            log.info("Finished TestServer execution - {} projects executed", execution.size());
+        } catch (IOException e) {
+            log.error("TestServer API call failed", e);
+            throw new AbortException("Could not start tests from project " + pathToProjectFile + " on server " + serverUrl);
+        } catch (ApiException e) {
+            log.error("TestServer API call failed when adding files", e);
```

Comment:
The message here is very specific but the ApiException can be thrown for more reasons. Are you sure it will be thrown only for pending files in this case?